### ISSUE TITLE
AttributeError on InterpSnapshotRZPotential

### ISSUE THREAD

**Issue by lepcampbell** [jobovy/galpy#248]
Hi Jo,

I'm working through the examples provided in the documentation (which are very helpful). Everything seems to be working great until I get to the new InterpSnapshotRZPotential example. 

Specifically, loading the interpolated snapshot:
>>>spi= InterpSnapshotRZPotential(h1,rgrid=(numpy.log(0.01),numpy.log(20.),101),logR=True,zgrid=(0.,10.,101),interpPot=True,zsym=True)

returns the following error:
    561     #Set up the C code
    562     ndarrayFlags= ('C_CONTIGUOUS','WRITEABLE')
--> 563     interppotential_calc_2dsplinecoeffs= _lib.samples_to_coefficients
    564     interppotential_calc_2dsplinecoeffs.argtypes= [ndpointer(dtype=numpy.float64,flags=ndarrayFlags),
    565                                                    ctypes.c_int,

AttributeError: 'NoneType' object has no attribute 'samples_to_coefficients'

Thanks in advance for any help you can provide.

**Comment by jobovy** [comment]
This looks to me like the correct C library was not loaded, probably because it wasn't installed. How did you install (pip, from source?) and did you get the C extensions installed (what was the output of the installation)? Did you get a warning message when running this code

    interppotential_c extension module not loaded, because XXX

?  If not, that would be weird, because if _lib is None, you should get this warning.

**Comment by lepcampbell** [comment]
Thanks for responding so quickly.

I installed with pip. My pip.log has been overwritten since then, so I don't know what the output of my installation was.

As you suspected, when I import potential from galpy I get the following:

from galpy import potential
galpyWarning: interppotential_c extension module not loaded, because galpy_actionAngle_c.so image was not found

**Comment by lepcampbell** [comment]
Thanks for responding so quickly.

I installed with pip. My pip.log has been overwritten since then, so I don't know what the output of my installation was.

As you suspected, when I import potential from galpy I get the following:

from galpy import potential
galpyWarning: interppotential_c extension module not loaded, because galpy_actionAngle_c.so image was not found

**Comment by jobovy** [comment]
Did the other C extensions get installed? I.e., do you get any other warnings when you do

    from galpy import orbit
    from galpy import actionAngle

If yes, then there is a problem with all of the C extensions, which probably means that you don't have the GSL. If no, then you might not have a recent enough version of the GSL; you can check your version with:
    
    gsl-config --version

This should be at least 1.14 for the code to work. If you fix any GSL issues, I think you probably have to first pip uninstall galpy and then re-install, although just using --upgrade might also work.

**Comment by lepcampbell** [comment]
The other C extensions were not installed. 

I updated the GSL to 1.16, uninstalled galpy, and now installing galpy fails. I attached the pip log.

---
Lauren Campbell, PhD
WIDER/CIRTL Postdoctoral Fellow
Center for Teaching
Vanderbilt University
________________________________________

**Comment by jobovy** [comment]
Resolved by upgrading the GSL and re-installing galpy.

**Comment by astrodietz** [comment]
I'm having a similar problem, maybe it's a simple fix but I haven't figured it out yet.
If I try:

from galpy import orbit

I get:

galpyWarning: interppotential_c extension module not loaded, because galpy_actionAngle_c.so image was not found
galpyWarning: integratePlanarOrbit_c extension module not loaded, because galpy_integrate_c.so image was not found
galpyWarning: integrateFullOrbit_c extension module not loaded, because galpy_integrate_c.so image was not found
galpyWarning: actionAngleAdiabatic_c extension module not loaded, because galpy_actionAngle_c.so image was not found
galpyWarning: actionAngleStaeckel_c extension module not loaded, because galpy_actionAngle_c.so image was not found

I've tried uninstalling/reinstalling/upgrading GSL and galpy multiple times but no luck.

**Comment by jobovy** [comment]
Hi,

This is most likely due to the fact that the installer cannot locate the GSL libraries (that was the original problem on this thread). You can tell the compiler where the GSL library is installed by defining environment variables::

    export CFLAGS=-I/usr/include
    export LDFLAGS=-L/usr/lib

or::

    setenv CFLAGS -I/usr/include
    setenv LDFLAGS -L/usr/lib

depending on your shell type (change the actual path to the include and lib directories that have the gsl directory). If you already have CFLAGS and LDFLAGS defined you just have to add the '-I/usr/include' and '-L/usr/lib' to them.

**Comment by jobovy** [comment]
Did this resolve your issue? If yes, please close the issue. If no, please give more information (e.g., send me the installation log).

**Comment by jobovy** [comment]
Closing this because of lack of updates and does not appear to be a real issue with the code.

**Comment by jobovy** [comment]
Did you download the TorusMapper code as explained [here](http://galpy.readthedocs.io/en/latest/installation.html#new-in-v1-2-installing-the-torusmapper-code)? If not, that's why you get that error message. All other functionality in galpy, including the other actionAngle modules will still work, only ``galpy.actionAngle.actionAngleTorus`` won't.

Did you run
```
from galpy import actionAngle
```
in the same session as the initial 
```
from galpy import orbit
```
? By default, warnings are only issued once.

**Comment by jcrestani** [comment]
Yes, I ran both imports in the same session. I just reinstalled galpy at home and got the (bright red) warning about TorusMapper during the installation. I don't know why it didn't show up in my lab computer. It works perfectly now, thank you! 

**I'll leave this here in case someone else has the same problem** (the gods know how many hours I lost trying to find out what was happening): initially, I got several "module not loaded". My previous codes didn't work at all. After reinstalling `libgsl0ldbl` and `libgsl0-dev` (necessary to run `gsl-config --version`), uninstalling galpy and reinstalling, there was a single "module not loaded" error left - actionAngleTorus_c. This was due to the missing TorusMapper code. Even without it, the rest works perfectly again. 

Thank you again for your time and for galpy, @jobovy !
